ts/main.ts

```diff
@@ -1,64 +1,92 @@
 interface Anime {
   mal_id: number;
   title: string;
+  synopsis: string;
   episodes: number;
   images: {
     jpg: {
       image_url: string;
     };
   };
-  [key: string]: any; // To account for any additional fields
+  rating: string;
 }
 
 const animeContainer = document.querySelector(
   '.anime-container',
 ) as HTMLDivElement;
 
-async function fetchAnime(): Promise<Anime[]> {
+async function fetchAnime(): Promise<void> {
   try {
     const response = await fetch('https://api.jikan.moe/v4/top/anime');
+    if (!response.ok) throw new Error('Fetch for top anime failed!');
     const data = await response.json();
-    return data.data as Anime[];
+
+    data.data.forEach((anime: Anime) => {
+      animeContainer.appendChild(renderAnime(anime));
+    });
   } catch (error) {
     console.error('Error fetching anime data:', error);
-    return [];
   }
 }
 
-function renderAnime(animeList: Anime[]): void {
-  animeContainer.innerHTML = '';
+// REnders the anime cards
+function renderAnime(anime: Anime): HTMLDivElement {
+  const animeDiv = document.createElement('div');
+  animeDiv.classList.add('animes');
 
-  animeList.forEach((anime) => {
-    const animeDiv = document.createElement('div');
-    animeDiv.classList.add('animes');
+  const imageDiv = document.createElement('div');
+  imageDiv.classList.add('image-container');
+  const img = document.createElement('img');
+  img.src = anime.images.jpg.image_url;
+  img.alt = anime.title;
+  imageDiv.appendChild(img);
 
-    const imageDiv = document.createElement('div');
-    imageDiv.classList.add('image-container');
-    const img = document.createElement('img');
-    img.src = anime.images.jpg.image_url;
-    img.alt = anime.title;
-    imageDiv.appendChild(img);
+  const textDiv = document.createElement('div');
+  textDiv.classList.add('text-container');
+  const title = document.createElement('h2');
+  title.textContent = anime.title;
+  const episodes = document.createElement('p');
+  episodes.textContent = `Episodes: ${anime.episodes ?? 'N/A'}`;
 
-    const textDiv = document.createElement('div');
-    textDiv.classList.add('text-container');
-    const title = document.createElement('h2');
-    title.textContent = anime.title;
-    const episodes = document.createElement('p');
-    episodes.textContent = `Episodes: ${anime.episodes ?? 'N/A'}`;
+  textDiv.appendChild(title);
+  textDiv.appendChild(episodes);
 
-    textDiv.appendChild(title);
-    textDiv.appendChild(episodes);
+  // Add an event listener to the title to swap views
+  title.addEventListener('click', () => {
+    swapView(anime);
+  });
 
```

Comment:
**For your next feature:** You should use event delegation rather than adding an event listener to each title. If you use delegation you will attach _one_ event listener to the parent container and you will need to conditionally check something (such as the `closest` element to have something unique that indicates it is a card)